src/java.base/share/classes/java/util/zip/ZipEntry.java

```diff
@@ -53,6 +53,7 @@
     long crc = -1;      // crc-32 of entry data
     long size = -1;     // uncompressed size of entry data
     long csize = -1;    // compressed size of entry data
+    boolean manual_csize = false; // Only true if csize was explicitely set by a call to setCompressedSize()
```

Comment:
Minor nit but this should probably be "csizeSet" as it's not too common to have underscore in field names.